graph-builder/src/plugins/openshift_secondary_metadata_parser/plugin.rs

```diff
@@ -0,0 +1,529 @@
+            sorted_releases.len()
+        );
+
+        Ok(io)
+    }
+}
+
+// Add some magic metadata which was manually added to quay
+fn add_manually_added_quay_metadata(graph: &mut cincinnati::Graph) -> Fallible<()> {
+    [
+        (
+            ("4.1.0-rc.7+amd64"),
+            vec![("io.openshift.upgrades.graph.previous.add", "4.1.0-rc.6")],
+        ),
+        (
+            ("4.1.0-rc.6+amd64"),
+            vec![
+                ("io.openshift.upgrades.graph.previous.add", "4.1.0-rc.5"),
+                // (
+                //     "io.openshift.upgrades.graph.release.channels",
+                //     "prerelease-4.1",
+                // ),
+            ],
+        ),
+        (
+            ("4.1.0-rc.8+amd64"),
+            vec![
+                ("io.openshift.upgrades.graph.previous.add", "4.1.0-rc.7"),
+                // (
+                //     "io.openshift.upgrades.graph.release.channels",
+                //     "prerelease-4.1",
+                // ),
+                //     "description": "",
+                //     "io.openshift.upgrades.graph.release.manifestref": "sha256:8250bbf79d4f567e24a7b85795aba97f9e75a9df18738891a0cb6ba42e422584",
+                //     "url": "https://access.redhat.com/errata/RHBA-2019:0758-09",
+                //     "io.openshift.upgrades.graph.release.arch": "amd64",
+            ],
+        ),
+        (
+            ("4.1.0-rc.9+amd64"),
+            vec![
+                (
+                    "io.openshift.upgrades.graph.previous.add",
+                    "4.1.0-rc.7,4.1.0-rc.8",
+                ),
+                // (
+                //     "io.openshift.upgrades.graph.release.channels",
+                //     "prerelease-4.1",
+                // ),
+                //     "description": "",
+                //     "io.openshift.upgrades.graph.release.arch": "amd64",
+                //     "io.openshift.upgrades.graph.release.manifestref": "sha256:49c4b6bf70061e522e3525aed534d087c9abfba7c39cbcbdd1bd770ab096bf9e",
+                //     "url": "https://access.redhat.com/errata/RHBA-2019:0758-09"
+            ],
+        ),
+        (
+            ("4.1.0+amd64"),
+            vec![
+                (
+                    "io.openshift.upgrades.graph.previous.add",
+                    "4.1.0-rc.4,4.1.0-rc.9",
+                ),
+                // (
+                //     "io.openshift.upgrades.graph.release.channels",
+                //     "prerelease-4.1,stable-4.1",
+                // ),
+            ],
+        ),
+        (
+            ("4.2.13-s390x+s390x"),
+            vec![
+                (
+                    "io.openshift.upgrades.graph.previous.add",
+                    "4.2.10-s390x,4.2.12-s390x",
+                ),
+                ("io.openshift.upgrades.graph.previous.remove", "*"),
+                (
+                    "io.openshift.upgrades.graph.release.channels",
+                    "candidate-4.2,fast-4.2,stable-4.2",
```

Comment:
This is [in cincinnati-graph-data][1] too.  Might be another parser bug?  The graph-data entries use the raw `4.2.13-s390x` name, which means they apply to all releases whose internal name is `4.2.13-s390x` regardless of image architecture.  However, there is only a single release image named `4.2.13-s390x` and it's an s390x image.  So cincinnati-graph-data could use either `4.2.13-s390x` or `4.2.13-s390x+s390x` to talk about that release, and the plugin should generate the same graph in both cases.

[1]: https://github.com/openshift/cincinnati-graph-data/blob/b70334977789366606fcec18ab4207870c6df279/channels/stable-4.2.yaml#L31

Reply:
The bug in this case was actually me putting this here. When I added the remaining parsing logic, this one stood out as duplicate channels in the assertion.